presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -408,6 +411,34 @@ protected Scope visitDelete(Delete node, Optional<Scope> scope)
             return createAndAssignScope(node, scope, Field.newUnqualified("rows", BIGINT));
         }
 
+        @Override
+        protected Scope visitAnalyze(Analyze node, Optional<Scope> scope)
+        {
+            analysis.setUpdateType("ANALYZE");
+            QualifiedObjectName tableName = createQualifiedObjectName(session, node, node.getTableName());
+
+            // verify the target table exists and it's not a view
+            if (metadata.getView(session, tableName).isPresent()) {
+                throw new SemanticException(NOT_SUPPORTED, node, "Analyzing views is not supported");
+            }
+
+            validateProperties(node.getProperties(), scope);
+            ConnectorId connectorId = metadata.getCatalogHandle(session, tableName.getCatalogName())
+                    .orElseThrow(() -> new PrestoException(NOT_FOUND, "Catalog not found: " + tableName.getCatalogName()));
+
+            Map<String, Object> analyzeProperties = metadata.getAnalyzePropertyManager().getProperties(
+                    connectorId,
+                    connectorId.getCatalogName(),
+                    mapFromProperties(node.getProperties()),
+                    session,
+                    metadata,
+                    analysis.getParameters());
+            analysis.setAnalyzeTarget(metadata.getTableHandleForAnalyze(session, tableName, analyzeProperties)
```

Comment:
You should check if user has read access to columns + insert access to table.
```suggestion
            TableHandle tableHandle = metadata.getTableHandleForAnalyze(session, tableName, analyzeProperties)
                .orElseThrow(() -> (new SemanticException(MISSING_TABLE, node, "Table '%s' does not exist", tableName))));
            // all column should be accessible by user to run ANALYZE 
            analysis.addTableColumnReferences(
                accessControl,
                session.getIdentity(),
                ImmutableMultimap.builder()
                  .putAll(tableName, metadata.getColumnHandles(session, tableHandle.get()).keys()
                  .build());
            analysis.setAnalyzeTarget(tableHandle);

            // make sure user has insert rights as ANALYZE changes table metadata
            // SHOULD WE HAVE EXPLICIT ANALYZE PRIVILEGE HERE? CAN ONE RUN ANALYZE IN HIVE WITHOUT
            // INSERT RIGHTS
            accessControl.checkCanInsertIntoTable(session.getRequiredTransactionId(), session.getIdentity(), tableName);
```

Please also:
* add test
* verify if one can run Analyze in Hive without "insert" privilege.